package-res/resources/web/pentaho/type/config/ConfigurationService.js

```diff
@@ -21,13 +21,150 @@ define([
 ], function(Base, SortedList, O, error) {
   "use strict";
 
+  /**
+   * List of names of context variables that are handled "generically" when sorting rules.
+   * More specific first.
+   *
+   * @type {string[]}
+   * @see pentaho.spec.IContextVars
+   * @see _ruleComparer
+   * @see _ruleFilterer
+   */
   var _selectCriteria = [
     "user",
     "theme",
     "locale",
     "application"
   ];
 
+  /**
+   * Map of merge operation name to operation handler function.
+   *
+   * @type {Object.<string, function>}
+   * @see mergeSpecsOne
+   */
+  var _mergeHandlers = {
+    "replace": mergeSpecsOperReplace,
+    "merge": mergeSpecsOperMerge,
+    "add": mergeSpecsOperAdd
+  };
+
+  /**
+   * The ordinal value of the next rule that is registered.
+   *
+   * This is used as the fallback rule order.
+   * Ensures sorting algorithm stability (as insertion order would be lost on resorts).
```

Comment:
Ensures sorting algorithm stability, because an insertion order would be lost during a re-sort.

**note to Dev: **
"re-sort" needs a hyphen because we are sorting again.

 "resort" means we are going on vacation somewhere nice, tropical, and with copious amounts of alcohol